.github/workflows/tests.yaml

```diff
@@ -0,0 +1,129 @@
+name: Spread tests
+on:
+  schedule:
+    # Weekly sunday testing for bitrot
+    - cron: '0 12 * * 0'
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: self-hosted
+    steps:
+      - name: Cleanup job workspace
+        id: cleanup-job-workspace
+        run: |
+          rm -rf "${{ github.workspace }}"
+          mkdir "${{ github.workspace }}"
+      - uses: actions/checkout@v2
+
+      - name: Build snap
+        run: |
+          spread -artifacts=./artifacts google-nested:tests/spread/build/
+          find ./artifacts -type f -name "*.deb" | head -n 1 | while read line ; do cp "$line" "${{ github.workspace }}/uc-initramfs.deb" ; done
+      
+      - uses: actions/upload-artifact@v2
+        with:
+          name: initramfsdeb
+          path: "${{ github.workspace }}/uc-initramfs.deb"
```

Comment:
It is not really safe in general to build a package in 22, and then install on 20. In this case, the dependencies for the current branch might not be available on ubuntu 20. So a safer thing to do would be to export the pc-kernel snap with initrd inject.